File: dotnet/DesignPatterns/StructuralDesignPatterns/CompositeDesignPattern/CompositeDesignPattern.Example2/IFileSystemItem.cs
namespace CompositeDesignPattern.Example2;

public abstract class IFileSystemItem
{
	public string Name { get; set; }
	public string Path { get; set; }
	public DateTime CreatedDate { get; set; }	

	public IFileSystemItem(string name, string path)
	{
		Name = name;
		Path = path;
		CreatedDate = DateTime.Now;
	}

	public abstract double GetSize();
}

File: java/DesignPatterns/StructuralDesignPatterns/CompositeDesignPattern/CompositeDesignPattern.Example2/src/main/java/structural/composite/example2/IFileSystemItem.java
package structural.composite.example2;

import java.time.LocalDateTime;

public abstract class IFileSystemItem {
    public String name;
    public String path;
    public LocalDateTime createdDate;

    public IFileSystemItem(String name, String path) {
        this.name = name;
        this.path = path;
        this.createdDate = LocalDateTime.now();
    }

    public abstract double getSize();
}
